File: examples/android/sync/app/src/main/java/com/mongodb/realm/examples/model/Dog.kt
package com.mongodb.realm.examples.model

import io.realm.RealmList
import io.realm.RealmObject
import io.realm.RealmResults
import io.realm.annotations.LinkingObjects
import io.realm.annotations.PrimaryKey
import org.bson.types.ObjectId

open class Dog(var name : String? = null): RealmObject() {
    @PrimaryKey
    var _id : ObjectId = ObjectId()
    var owner: RealmList<Person>? = null
}

File: examples/react-native/__tests__/js/models/Dog.js
import Realm from 'realm';

// :snippet-start: js-dog-schema
class Dog extends Realm.Object {
  static schema = {
    name: 'Dog',
    properties: {
      name: 'string',
      owner: 'Person?',
      age: 'int?',
    },
  };
}
// :snippet-end:
export default Dog;
